backend/src/v2/compiler/argocompiler/dag.go

```diff
@@ -48,7 +49,18 @@ func (c *workflowCompiler) DAG(name string, componentSpec *pipelinespec.Componen
 		},
 		DAG: &wfapi.DAGTemplate{},
 	}
-	for taskName, kfpTask := range dagSpec.GetTasks() {
+	tasks := dagSpec.GetTasks()
+	// Iterate through tasks in deterministic order to facilitate testing.
+	// Note, order doesn't affect compiler with real effect right now.
+	// In the future, we may consider using topology sort when building local
+	// executor that runs on pipeline spec directly.
+	keys := make([]string, 0, len(tasks))
+	for key := range tasks {
+		keys = append(keys, key)
+	}
```

Comment:
I think this chunk can be just

```golang
keys := dagSpec.GetTasks().Keys()
```

Reply:
My bad, ignore me then.

Reply:
With 
> ```go
> keys := reflect.ValueOf(dagSpec.GetTasks()).MapKeys()
> ```
It doesn't feel very readable? it also introduces a new package. The variable `tasks` is used again at line #58 so I don't think it makes a huge difference.